### ISSUE TITLE
Enable virtual fields

### ISSUE THREAD

**Issue by christoffkok** [hiho-/SQLEDITABLE#1]
Hi,

I'm not sure if this is the place you'd want for reporting feature requests.

Feature Requests:
- Make SQLEditable work with virtual fields.
- Make SQLEditable work with web2py versioning.

Virtual fields are displayed in the web2py smartgrid, but not in SQLEditable.
Currently I get an error when I try to delete a record. It is removed from the view if I refresh though.

**Comment by hiho-** [comment]
Thanks for your request.
You use 'TABLE.virtualfields.append' method, don't you ?.
I will check the code. Please wait.

**Comment by hiho-** [comment]
Hi.
I changed for support the virtual field. @b943faca5d41cab5962082a1bf335a026aec8638

Please see to https://github.com/hiho-/SQLEDITABLE/releases/tag/0.14 .
You can use 'header' parameter. 'demo090'. http://sqleditable.appspot.com/demo090
And, if you want to versioning, you can use '_enable_record_versioning()' to the table.

``` python
class san(object):
    def san(self):
        return self.employee_sheet.name + '-san'
db.employee_sheet.virtualfields.append(san())
```

``` python
db.employee_sheet.san2 = Field.Virtual(lambda row: row.employee_sheet.name + '-SAN')
```

``` python
def demo090():    
    response.title = 'demo090'
    response.view = 'plugin_sqleditable/sample.html'
    header = ['employee_number','name','san','san2'] 
    editable = SQLEDITABLE(db.employee_sheet, header=header, showid=False, maxrow=5, deletable=True).process()
    return dict(editable=editable)
```

**Comment by christoffkok** [comment]
Thank you Hitoshi Kato,

Thanks for the effort. I appreciate it.

Regards,

On 20 January 2015 at 17:02, Hitoshi Kato notifications@github.com wrote:

> Closed #1 https://github.com/hiho-/SQLEDITABLE/issues/1.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hiho-/SQLEDITABLE/issues/1#event-221644865.

## 

Christoff Kok
Software Engineer
Ex Mente

http://www.ex-mente.co.za
christoff.kok@ex-mente.co.za
PO Box 10214
Centurion
0046
South Africa
tel: +27 12 743 6993
tel: +27 12 654 8198
fax: +27 85 150 1341
